compositor_render/src/renderer/texture/base.rs

```diff
@@ -31,7 +36,11 @@ impl Texture {
 
         let view = texture.create_view(&wgpu::TextureViewDescriptor::default());
 
-        Self { texture, view }
+        Self {
+            texture,
+            view,
+            download_buffer: None,
```

Comment:
We don't want download buffers to be a part of texture, because not all textures need them. Using Option avoids unnecessary entities, but adding them to texture is a hack.

It's fine if e.g. texture has an implementation for creating download buffer for that texture, but buffer itself should be stored separately. Check how it's implemented in scene.